File: src/main/org/nlogo/api/LogoException.java
package org.nlogo.api;

/**
 * <p>A runtime error that has occurred in NetLogo code.
 * <p/>
 * <p>As with any Exception, use the getMessage() method to get the error message.
 */

// LogoExceptions are expected to have nice end-user-understandable
// descriptions

// LogoException is abstract because engine code is supposed to throw
// a concrete subclass like EngineException. - ST 5/4/10

public abstract strictfp class LogoException
    extends Exception {

  public LogoException(String message) {
    super(message);
  }

  public LogoException(String message, Throwable cause) {
    super(message, cause);
  }

}

File: headless/src/main/org/nlogo/api/LogoException.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.api

/**
 * A runtime error that has occurred in NetLogo code.
 *
 * As with any Exception, use the getMessage() method to get the error message.
 * A LogoExceptions is expected to have a nice end-user-understandable message.
 *
 * LogoException is abstract because engine code is supposed to throw
 * a concrete subclass like EngineException.
 */

abstract class LogoException(message: String, cause: Throwable = null)
extends Exception(message, cause)
